pageserver/src/tenant/timeline/init.rs

```diff
@@ -201,25 +203,15 @@ pub(super) fn cleanup(path: &Utf8Path, kind: &str) -> anyhow::Result<()> {
     std::fs::remove_file(path).with_context(|| format!("failed to remove {kind} at {path}"))
 }
 
-pub(super) fn cleanup_local_file_for_remote(
-    local: &LocalLayerFileMetadata,
-    remote: &LayerFileMetadata,
-) -> anyhow::Result<()> {
-    let local_size = local.metadata.file_size();
-    let remote_size = remote.file_size();
+pub(super) fn cleanup_local_file_for_remote(local: &LocalLayerFileMetadata) -> anyhow::Result<()> {
```

Comment:
Is this still the case or not, the generation numbers are now part of the file identity, so why is this not a case of "local only"?

Reply:
Right now, we don't parse the generation number from the local path: it's just used opaquely to generate a more-unique path.

We could do that to disambiguate a bad file from a different generation (if the local file is new enough to have a generation), although currently the action we take is the same :thinking:

Reply:
Ok, I was thinking it was implemented in some other fashion it ended up being implemented. Shame on me for not building the test case which I thought was so easy to build :)

In the implementation I had in mind, it would be a matter of noticing the other file has no generation and other local file has a generation... But that is not simple either. Perhaps this works, I should now re-read it.